ui/components/app/permission-page-container/permission-page-container-content/permission-page-container-content.component.js

```diff
@@ -33,9 +42,29 @@ export default class PermissionPageContainerContent extends PureComponent {
     const { t } = this.context;
 
     const items = Object.keys(selectedPermissions).map((permissionName) => {
-      const description = t(permissionName);
-      // don't allow deselecting eth_accounts
-      const isDisabled = permissionName === 'eth_accounts';
+      const isEthAccounts = permissionName === 'eth_accounts';
+
+      let description = `Unknown permission: "${permissionName}"`;
+      if (isEthAccounts) {
+        description = t(permissionName);
+      }
+      ///: BEGIN:ONLY_INCLUDE_IN(flask)
+      else if (permissionName.startsWith(invokeSnapPrefix)) {
+        description = t(invokeSnapPrefix, [
+          permissionName.replace(invokeSnapPrefix, ''),
+        ]);
+      } else if (permissionName.startsWith(getBip44EntropyPrefix)) {
+        // TODO:flask create coin_type to protocol name enum
+        description = t(getBip44EntropyPrefix, [
+          permissionName.replace(getBip44EntropyPrefix, ''),
+        ]);
+      } else {
+        description = t(permissionName);
+      }
```

Comment:
@Gudahtt, I decided to continue using permission names (or, in the case of namespaced permissions, target keys without `*`, since that character apparently cannot appear as top-level object keys in extension JSON files) as the locale message key for permission descriptions, but with this implementation everything except `getBip44Entropy` (which appears as a string literal in this file) is flagged as unused by the locale message verification script. How do you recommend we deal with this?

Reply:
That script requires the permission names to be _somewhere_ on disk at build time. That should be the case here too, right?

It would be easiest if we could use string-literals here somehow. We can direct the script to look elsewhere in special cases though.

Reply:
Yeah, I was thinking I could create a utility function with a switch statement. I'll get to it!

Reply:
Done here: 406ef67

Reply:
Sorry I don't follow how that solves the issue. This still isn't compatible with the strict mode of verify locales, because we aren't passing string literals to the `t` function.

Though it looks like all the strings we need have been laid out now at least. We can switch to using a mapping of enums to messages, so that the linter can find these messages more easily. Like we did here for example: https://github.com/MetaMask/metamask-extension/blob/develop/ui/pages/settings/advanced-tab/advanced-tab.component.js#L453

Reply:
Ah, I didn't run the strict version of the script locally. I'll get to it!

Reply:
Although, I have to say that I am not a fan of the strict version of that script, because it's forcing me to use magic strings instead of enums. We should consider creating message enums that can be interpreted by that script so that we can use enums without having to do AST analysis.